lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php

```diff
@@ -296,7 +301,11 @@ public function delete($entity)
             $parentTable    = $this->quoteStrategy->getTableName($parentMetadata, $this->platform);
             $parentTypes    = $this->getClassIdentifiersTypes($parentMetadata);
 
-            $this->conn->delete($parentTable, $id, $parentTypes);
+            // Fix for bug GH-8229 (id column from parent class renamed in child class):
+            // Use the correct name for the id column as named in the parent class.
+            $parentId = array_combine($parentMetadata->getIdentifierColumnNames(), $identifier);
```

Comment:
That depends on the platform you use. If you use a database with foreign key support you're right, then you won't get there. If you use SQLite instead, the test would fail there.

IMHO the whole block starting in line 281 ("if ($this->platform->supportsForeignKeyConstraints()) {") should be removed, because this is a potential bug: If a platform supports foreign keys in general (or you decide to support foreign keys for a platform later), the entries are only deleted from the parent table, assuming that the foreign keys are really set and handle the deletion of the entries from the child table, but it's never checked if they exist. So if the foreign key is missing the entry in the child table will never be deleted (and perhabs result in other errors).

Removing this prevent this potential error, and the test would always fail without the change.  

WDYT?

Reply:
There was another potential bug: When a foreign key was set but not expected, because not supported by DBAL, i.e. when you use an existing SQLite database - SQLite supports foreign keys, by DBAL doesn't (for good reason I think, because not updateable). In this case the deletion was done in wrong order (child -> root) and would have failed.

I refactored this method to remove the previous duplicate code and the resulting duplicate bug fix as well as the two potential bugs.